lib/modules/home/components/body.dart

```diff
@@ -6,61 +6,78 @@ class Body extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final state = context.findAncestorStateOfType<_HomeViewImplState>()!;
+
     return StreamsSelector0<bool>.value(
       stream: state.isProgressHUDShown,
       builder: (_, isProgressHUDShown, child) => ProgressHUD(
         isShow: isProgressHUDShown,
         child: child!,
       ),
-      child: Screen(
-        body: SimpleGestureDetector(
-          onVerticalSwipe: (direction) {
-            if (direction != SwipeDirection.down) return;
-            state.delegate?.didSwipeDown.add(null);
-          },
-          child: Container(
-            color: Colors.black,
-            child: StreamsSelector0<bool>.value(
-              stream: state._isLoading,
-              builder: (_, isLoading, child) => Skeleton(
-                key: HomeView.skeletonKey,
-                isShown: isLoading,
-                baseColor: Colors.white.withOpacity(0.12),
-                highlightColor: Colors.white.withOpacity(0.5),
-                child: child!,
-              ),
-              child: Stack(
-                fit: StackFit.expand,
-                children: [
-                  StreamsSelector0<List<SurveyInfo>>.value(
-                    stream: state._surveys,
-                    builder: (_, surveys, __) => surveys.isNotEmpty
-                        ? CarouselSlider.builder(
-                            carouselController: state._pageController,
-                            itemCount: surveys.length,
-                            itemBuilder: (_, index, ___) => SlideItem(
-                              survey: surveys[index],
-                            ),
-                            options: CarouselOptions(
-                              viewportFraction: 1,
-                              initialPage: state._currentPage.value,
-                              height: double.infinity,
-                              onPageChanged: (index, _) =>
-                                  state._currentPage.add(index),
-                            ),
-                          )
-                        : SlideItem.empty(),
+      child: SliderMenuContainer(
+        key: state._sliderMenuContainerKey,
+        hasAppBar: false,
+        drawerIcon: const SizedBox.shrink(),
+        slideDirection: SlideDirection.RIGHT_TO_LEFT,
+        sliderMenu: locator.get<SideMenuModule>(),
```

Comment:
any reason why do we use DI here? 🙏  Can't we create `SideMenuModule()` directly? 💭  I don't think we should apply DI to widget tree 😅

Reply:
@manh-t  So that, when doing a test for Home screen, I can stub this and return an empty widget 😬

Reply:
why do we need to stub this widget? 😅 
I can say, in Flutter widget is everything, and widget can rebuild any times. So putting DI to widget tree is error-prone, because if we cannot manage DI well, then we may face issue that we need the widget to be rebuilt but it's not.
I recommend to define normal constructor like others. 
```suggestion
        sliderMenu: SideMenuModule(),
```

Reply:
@markgravity I think you will need to fix this, please don't apply DI to widget tree 🙇

Reply:
@manh-t I updated it 🙏